// @vercel/sdk/esm/sdk/environment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { environmentDeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId } from "../funcs/environmentDeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId.js";
import { environmentGetV9ProjectsIdOrNameCustomEnvironments } from "../funcs/environmentGetV9ProjectsIdOrNameCustomEnvironments.js";
import { environmentGetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId } from "../funcs/environmentGetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId.js";
import { environmentPatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId } from "../funcs/environmentPatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId.js";
import { environmentPostV1ProjectsIdOrNameCustomEnvironments } from "../funcs/environmentPostV1ProjectsIdOrNameCustomEnvironments.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class Environment extends ClientSDK {
    /**
     * Create a custom environment for the current project.
     *
     * @remarks
     * Creates a custom environment for the current project. Cannot be named 'Production' or 'Preview'.
     */
    async postV1ProjectsIdOrNameCustomEnvironments(request, options) {
        return unwrapAsync(environmentPostV1ProjectsIdOrNameCustomEnvironments(this, request, options));
    }
    /**
     * Retrieve custom environments
     *
     * @remarks
     * Retrieve custom environments for the project. Must not be named 'Production' or 'Preview'.
     */
    async getV9ProjectsIdOrNameCustomEnvironments(request, options) {
        return unwrapAsync(environmentGetV9ProjectsIdOrNameCustomEnvironments(this, request, options));
    }
    /**
     * Retrieve a custom environment
     *
     * @remarks
     * Retrieve a custom environment for the project. Must not be named 'Production' or 'Preview'.
     */
    async getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId(request, options) {
        return unwrapAsync(environmentGetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId(this, request, options));
    }
    /**
     * Update a custom environment
     *
     * @remarks
     * Update a custom environment for the project. Must not be named 'Production' or 'Preview'.
     */
    async patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId(request, options) {
        return unwrapAsync(environmentPatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId(this, request, options));
    }
    /**
     * Remove a custom environment
     *
     * @remarks
     * Remove a custom environment for the project. Must not be named 'Production' or 'Preview'.
     */
    async deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId(request, options) {
        return unwrapAsync(environmentDeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrId(this, request, options));
    }
}
