// @vercel/sdk/esm/models/deletev9projectsidornamecustomenvironmentsenvironmentslugoridop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema = z.object({
    deleteUnassignedEnvironmentVariables: z.boolean().optional(),
});
/** @internal */
export const DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema = z.object({
    deleteUnassignedEnvironmentVariables: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$;
(function (DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$) {
    /** @deprecated use `DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema` instead. */
    DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$.inboundSchema = DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema;
    /** @deprecated use `DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema` instead. */
    DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$.outboundSchema = DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema;
})(DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$ || (DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$ = {}));
export function deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBodyToJSON(deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody) {
    return JSON.stringify(DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema
        .parse(deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody));
}
export function deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody' from JSON`);
}
/** @internal */
export const DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema = z.object({
    idOrName: z.string(),
    environmentSlugOrId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema = z.object({
    idOrName: z.string(),
    environmentSlugOrId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$;
(function (DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$) {
    /** @deprecated use `DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema` instead. */
    DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$.inboundSchema = DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema;
    /** @deprecated use `DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema` instead. */
    DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$.outboundSchema = DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema;
})(DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$ || (DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$ = {}));
export function deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestToJSON(deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest) {
    return JSON.stringify(DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema
        .parse(deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest));
}
export function deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest' from JSON`);
}
/** @internal */
export const DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema = z.object({});
/** @internal */
export const DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$;
(function (DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$) {
    /** @deprecated use `DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema` instead. */
    DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$.inboundSchema = DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema;
    /** @deprecated use `DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema` instead. */
    DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$.outboundSchema = DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema;
})(DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$ || (DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$ = {}));
export function deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBodyToJSON(deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody) {
    return JSON.stringify(DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema
        .parse(deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody));
}
export function deleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'DeleteV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody' from JSON`);
}
