// @vercel/sdk/esm/models/getv9projectsidornamecustomenvironmentsenvironmentslugoridop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema = z.object({
    idOrName: z.string(),
    environmentSlugOrId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema = z.object({
    idOrName: z.string(),
    environmentSlugOrId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$;
(function (GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$) {
    /** @deprecated use `GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema` instead. */
    GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$.inboundSchema = GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema;
    /** @deprecated use `GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema` instead. */
    GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$.outboundSchema = GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema;
})(GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$ || (GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$ = {}));
export function getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestToJSON(getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest) {
    return JSON.stringify(GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema
        .parse(getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest));
}
export function getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest' from JSON`);
}
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema = z.object({});
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$;
(function (GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$) {
    /** @deprecated use `GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema` instead. */
    GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$.inboundSchema = GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema;
    /** @deprecated use `GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema` instead. */
    GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$.outboundSchema = GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema;
})(GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$ || (GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$ = {}));
export function getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBodyToJSON(getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody) {
    return JSON.stringify(GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema
        .parse(getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody));
}
export function getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody' from JSON`);
}
