// @vercel/sdk/esm/models/patchv9projectsidornamecustomenvironmentsenvironmentslugoridop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * Type of matcher. One of \"equals\", \"startsWith\", or \"endsWith\".
 */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType = {
    Equals: "equals",
    StartsWith: "startsWith",
    EndsWith: "endsWith",
};
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$inboundSchema = z.nativeEnum(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType);
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$outboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$;
(function (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$) {
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$inboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$.inboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$inboundSchema;
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$outboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$.outboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$outboundSchema;
})(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$ || (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$ = {}));
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$inboundSchema = z.object({
    type: PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$inboundSchema,
    pattern: z.string(),
});
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$outboundSchema = z.object({
    type: PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdType$outboundSchema,
    pattern: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$;
(function (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$) {
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$inboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$.inboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$inboundSchema;
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$outboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$.outboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$outboundSchema;
})(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$ || (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$ = {}));
export function patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcherToJSON(patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher) {
    return JSON.stringify(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$outboundSchema
        .parse(patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher));
}
export function patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcherFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher' from JSON`);
}
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema = z.object({
    slug: z.string().optional(),
    description: z.string().optional(),
    branchMatcher: z.nullable(z.lazy(() => PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$inboundSchema)).optional(),
});
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema = z.object({
    slug: z.string().optional(),
    description: z.string().optional(),
    branchMatcher: z.nullable(z.lazy(() => PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdBranchMatcher$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$;
(function (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$) {
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$.inboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema;
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$.outboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema;
})(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$ || (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$ = {}));
export function patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBodyToJSON(patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody) {
    return JSON.stringify(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema
        .parse(patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody));
}
export function patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody' from JSON`);
}
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema = z.object({
    idOrName: z.string(),
    environmentSlugOrId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema = z.object({
    idOrName: z.string(),
    environmentSlugOrId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestBody$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$;
(function (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$) {
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$.inboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema;
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$.outboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema;
})(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$ || (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$ = {}));
export function patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestToJSON(patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest) {
    return JSON.stringify(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$outboundSchema
        .parse(patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest));
}
export function patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdRequest' from JSON`);
}
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema = z.object({});
/** @internal */
export const PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$;
(function (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$) {
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$.inboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema;
    /** @deprecated use `PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema` instead. */
    PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$.outboundSchema = PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema;
})(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$ || (PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$ = {}));
export function patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBodyToJSON(patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody) {
    return JSON.stringify(PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$outboundSchema
        .parse(patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody));
}
export function patchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PatchV9ProjectsIdOrNameCustomEnvironmentsEnvironmentSlugOrIdResponseBody' from JSON`);
}
